test/e2e/common.go

```diff
@@ -359,3 +360,56 @@ func IsBootedFromDisk(client *ssh.Client) (bool, error) {
 
 	return isDisk, nil
 }
+
+type BuildAndInstallDeploymentInput struct {
+	Kustomization       string
+	ClusterProxy        framework.ClusterProxy
+	WaitForDeployment   bool
+	WatchDeploymentLogs bool
+	DeploymentName      string
+	DeployNS            string
+	LogPath             string
+	WaitIntervals       []interface{}
+}
+
+func BuildAndInstallDeployment(ctx context.Context, input *BuildAndInstallDeploymentInput) error {
```

Comment:
Could you please add a doc string?

I think we can make it slightly more generic still, which should make it useful for even more things. We are really building and applying a kustomization, so I think changing the name would make sense.
```suggestion
func BuildAndApplyKustomization(ctx context.Context, input *BuildAndInstallDeploymentInput) error {
```

Reply:
Thanks. I thought of something similar in the first place, but the "watch" stuff would look very weird if what we install doesn't have any deployments :sweat_smile: 
Maybe we can make it work with some explanation in the docs? I agree that later on we could use this for other things as well.

Reply:
Yeah but we can actually use it for stuff without deployment already as long as we do not enable wait/watch :slightly_smiling_face: 
Explanations in the doc string (for either the function or the struct) is always good!